tests/cupy_tests/manipulation_tests/test_kind.py

```diff
@@ -10,6 +10,20 @@
 @testing.gpu
 class TestKind(unittest.TestCase):
 
+    @testing.for_all_dtypes()
+    def test_asarray_chkfinite(self, dtype):
+        a = [0, 4, 0, 5]
+        a_gpu = cupy.asarray_chkfinite(a, dtype=dtype)
+        a_cpu = numpy.asarray_chkfinite(a, dtype=dtype)
+        assert a_gpu.dtype == a_cpu.dtype
+
+    @testing.for_all_dtypes()
+    def test_asarray_chkfinite_non_finite_vals(self, dtype):
+        a = [-numpy.inf, 0., numpy.inf]
+        a_gpu = cupy.asarray_chkfinite(a, dtype=dtype)
```

Comment:
I believe this is what you need @khushi-411 

```suggestion
        self.assertRaises(ValueError, cupy.asarray_chkfinite, a, dtype):
```

This will check that the expected ValueError is raised when you are passing an array with infinite values.

Reply:
Thanks a lot, @oke-aditya, for taking a look (: It worked.
But I am not too sure if we need this test or not.

Reply:
I believe you need the test. It is recommended to test the valueErrors as this is an expected behivour.

Any case the maintainers will tell you :)

Reply:
Please see my review comments below.